### DISCUSSION TITLE
Sideloading notification with paid dev account

### DISCUSSION THREAD

**Discussion by Shura1oplot** [SoCuul/SCInsta#51] [Support] [answered]
The description mentioned that while having account with enrolled Apple Developer Program it is possible to have notifications in a sideloaded app.

> Sideloading
> ...
> [Sideloadly](https://sideloadly.io/#download) (Free, No notifications*) *Notifications require $99/year Apple Developer Program
> 

I have an Apple Developer Program enrolled, but don't have notifications. Are there any changes to be made in Sideloadly to have notifications?

**Comment by SoCuul** [comment]
There's a long guide you have to go through haha. I'll post a guide when I'm free

**Comment by Shura1oplot** [comment]
Thank you! Much appreciated. It'll help me a lot.

**Comment by SoCuul** [comment, marked as answer]
## Register app id
1. Register an identifier with the "App IDs" tag identifier at https://developer.apple.com/account/resources/identifiers/add/bundleId
2. Enter a description of your choice
3. Set the bundle id to "explicit", with a value of your choice. For example: com.socuul.youtube
4. Enable the "Push Notifications" capability, along with any other ones you would like
5. Click the continue button, then register

## Generate Certificate Signing Request
6. Open the `Keychain Access` app on your mac.
7. Under the **Keychain Access** entry on your menu bar, select Certificate Assistant > Request a Certificate from a Certificate Authority
8. Enter an email address and a name of your choice (the contents does not matter)
9. Select **saved to disk**
10. Save this file in a folder of your choice

## Create distribution certificate
11. Create a new "Apple Distribution" certificate under the **Software** header at https://developer.apple.com/account/resources/certificates/add
12. Upload the Certificate Signing Request file you generated in the previous section
13. Click the **Download** button, and save the `.cer` file in a folder of your choice
14. Open the downloaded file, and make sure you can find it inside of your keychain.
![](https://i.imgur.com/QJfkFEF.png)
15. Right click on the certificate and select **Export**
16. Make sure the file format is set to `.p12`, and save the file in a folder of your choice

## Create Ad Hoc profile
17. Create a new "Ad Hoc" profile under the **Distribution** header at https://developer.apple.com/account/resources/profiles/add
18. Select the app id you created earlier
19. Make sure to select the **no** option for offline support
20. Select the distribution certificate you created in the previous section
21. Select which devices you will install the app on
22. Provide a name to easily identify the profiler later, then click the **Generate** button.
23. Click the **Download** button, and save the `.mobileprovision` file in a folder of your choice

## Install tools
24. Make sure you have an installation of [homebrew](https://brew.sh), otherwise install it with `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`
25. Install OpenSSL & zsign
```sh
$ brew install openssl@1.1

$ git clone https://github.com/zhlynn/zsign
$ cd zsign
$ chmod +x INSTALL.sh
$ ./INSTALL.sh
```

## Export certificate information
26.  Run the following commands, replacing <yourcert.p12> with the location of your saved p12 certificate file.
```sh
$ openssl pkcs12 -in yourcert.p12 -nocerts -nodes -legacy -out ios_dist.key
$ openssl pkcs12 -in yourcert.p12 -clcerts -nokeys -legacy -out ios_dist.pem
```

## Sign app
27. Run the following command and replace the following values.
	- <dist.cer> → Your distribution certificate file path
	- <adhoc.mobileprovision> → Your ad hoc profile path
	- <bundle.id> → The bundle id of your chosen app (eg: com.hammerandchisel.discord)
	- <app.name> → The name to display on the home screen (eg: Discord)
	- <signed.ipa> → The name to save the signed ipa under
	- <unsigned.ipa> → The path to the unsigned ipa you would like to sign

```sh
$ ./zsign -z 5 -k ios_dist.key -c <dist.cer> -m <adhoc.mobileprovision> -b <bundle.id> -n <app.name> -o <signed.ipa> <unsigned.ipa>
```
## Install app
28. Finally, install the signed `.ipa` to your device through your preferred method. For example:
	- Sending `.ipa` to device through AirDrop
	- Dragging `.ipa` onto connected device in finder
	- Dragging `.ipa` onto connected device in Apple Configurator
	- Using a command-line tool such as https://libimobiledevice.org/

**Comment by SoCuul** [comment]
Basically, you won't be using sideloadly at all. After it's signed you can just send it to your phone by any means (airdrop, finder over usb, even safari)

**Comment by Shura1oplot** [comment]
@SoCuul, I was able to sign and install an app, but push notifications doesn't work :( Probably some steps are missing, because the same app installed using the same dev account through appdb supports push notifications. How can I diagnose why there is no push?

**Reply by Shura1oplot** [reply]
Problem solved, you need to use "com.burbn.instagram" as a bundle id in zsign commands line.

**Reply by SoCuul** [reply]
Glad you got it working!
